### ISSUE TITLE
Handle --port=0

### ISSUE THREAD

**Issue by ghost** [svenstaro/miniserve#81]
The current behaviour of `miniserve` if set to `--port=0` is to accept it (valid `u16`). However, this is not a "valid" port (actually, it is valid, but shouldn't be used as such), an address like `127.0.0.1:0` is not valid .

Chrome's message:
```
This site can’t be reached
The web page at http://127.0.0.1:0/ might be temporarily down or it may have moved permanently to a new web address.
ERR_ADDRESS_INVALID
```

By convention (I don't know if there's a spec for it ?), addressing a system on port 0 means "give me whatever port you want", usually from 49152 to 65535.

So, we have 2 solutions here:
- either we make the port 0 invalid, exception caught at startup, no big deal
- or we implement such a feature if you think that's worth it: generate a random port from 49152 to 65535 and use it. Note that Python's `http.server` implements this feature

In short, either a bug fix, or a new feature :)

**Comment by svenstaro** [comment]
Good catch! I'm definitely for option 2. To find a free port, we can use `port_check`, which we already use for the tests.
